jerry-core/jerry.c

```diff
@@ -1698,6 +1698,8 @@ jerry_parse (const jerry_api_char_t *source_p, /**< script source */
 {
   jerry_assert_api_available ();
 
+  vm_finalize ();
```

Comment:
That's actually "finalize if it has already been initialized"; the check happens within `vm_finalize`. If this call is omitted, `jerry_parse` cannot be invoked multiple times.